lib/Serialization/Deserialization.cpp

```diff
@@ -166,10 +167,29 @@ static void skipRecord(llvm::BitstreamCursor &cursor, unsigned recordKind) {
 }
 
 void ModuleFile::fatal(llvm::Error error) const {
+  Core->fatal(diagnoseFatal(std::move(error)));
+}
+
+llvm::Error ModuleFile::diagnoseFatal(llvm::Error error) const {
   if (FileContext)
     getContext().Diags.diagnose(SourceLoc(), diag::serialization_fatal,
                                 Core->Name);
-  Core->fatal(std::move(error));
+  // Crash. ModuleFileSharedCore::fatal() calls abort(). This allows
+  // aggregation of crash logs for compiler development, but in a
+  // long-running process like LLDB this is undesirable. Only abort()
+  // if not in the debugger.
+  if (!getContext().LangOpts.DebuggerSupport)
+    Core->fatal(std::move(error));
```

Comment:
With this version I believe we'll still have make sure that our crash log aggregator ignores `ModuleFile::diagnoseFatal` . It needs to ignore all the functions on top of the stack at a time of the crash in order to record the crash location as being wherever called `ModuleFile::diagnoseFatal`. Otherwise, in this state, our aggregator will report all deserialization crash as being on this precise line.